src/dodal/common/beamlines/beamline_utils.py

```diff
@@ -170,3 +179,6 @@ def get_path_provider() -> PathProvider:
 def clear_path_provider() -> None:
     global PATH_PROVIDER
     del PATH_PROVIDER
+
+
+set_path_provider(DEFAULT_PATH_PROVIDER)
```

Comment:
Why set and not just assign to `PATH_PROVIDER`? If it's set to `PATH_PROVIDER` it'll be gc'd if it's ever overwritten.